File: OneSignalSDK/onesignal/src/main/java/com/onesignal/outcomes/model/OSOutcomeSourceBody.java
package com.onesignal.outcomes.model;

import android.support.annotation.Nullable;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

public class OSOutcomeSourceBody {

    private static final String NOTIFICATION_IDS = "notification_ids";
    private static final String IAM_IDS = "in_app_message_ids";

    @Nullable
    private JSONArray notificationIds;
    @Nullable
    private JSONArray inAppMessagesIds;

    public OSOutcomeSourceBody() {
        this(new JSONArray(), new JSONArray());
    }

    public OSOutcomeSourceBody(@Nullable JSONArray notificationIds, @Nullable JSONArray inAppMessagesIds) {
        this.notificationIds = notificationIds;
        this.inAppMessagesIds = inAppMessagesIds;
    }

    public JSONObject toJSONObject() throws JSONException {
        JSONObject json = new JSONObject();
        json.put(NOTIFICATION_IDS, notificationIds);
        json.put(IAM_IDS, inAppMessagesIds);
        return json;
    }

    @Nullable
    public JSONArray getNotificationIds() {
        return notificationIds;
    }

    public void setNotificationIds(@Nullable JSONArray notificationIds) {
        this.notificationIds = notificationIds;
    }

    @Nullable
    public JSONArray getInAppMessagesIds() {
        return inAppMessagesIds;
    }

    public void setInAppMessagesIds(@Nullable JSONArray inAppMessagesIds) {
        this.inAppMessagesIds = inAppMessagesIds;
    }

    @Override
    public String toString() {
        return "OSOutcomeSourceBody{" +
                "notificationIds=" + notificationIds +
                ", inAppMessagesIds=" + inAppMessagesIds +
                '}';
    }
}

File: OneSignalSDK/onesignal/src/main/java/com/onesignal/outcomes/domain/OSOutcomeSourceBody.kt
package com.onesignal.outcomes.domain

import com.onesignal.outcomes.OSOutcomeConstants
import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

class OSOutcomeSourceBody @JvmOverloads constructor(var notificationIds: JSONArray? = JSONArray(), var inAppMessagesIds: JSONArray? = JSONArray()) {

    @Throws(JSONException::class)
    fun toJSONObject(): JSONObject = JSONObject()
            .put(OSOutcomeConstants.NOTIFICATION_IDS, notificationIds)
            .put(OSOutcomeConstants.IAM_IDS, inAppMessagesIds)

    override fun toString(): String {
        return "OSOutcomeSourceBody{" +
                "notificationIds=" + notificationIds +
                ", inAppMessagesIds=" + inAppMessagesIds +
                '}'
    }
}